File: basic_arithmetic/sum/sum.ts
export const sum = (a: number, b: number): number => {
    return a + b;
}

interface Sum{
    a: number;
    b: number;
}

export const sumWithInterface = (sum: Sum): number => sum.a + sum.b;

export class SumClass{
    a: number;
    b: number;

    constructor(a: number, b: number){
        this.a = a;
        this.b = b;
    }

    sum(): number{
        return this.a + this.b;
    }

    static sum(a: number, b: number): number{
        return a + b;
    }

    static sumWithInterface(sum: Sum): number{
        return sum.a + sum.b;
    }
}

File: basic_arithmetic/sum/Sum.java
package basic_arithmetic.sum;
import java.util.Scanner;

public class Sum{


    public int SumNumbers(){  
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the first number: ");
        int firstNumber = scanner.nextInt();

        System.out.print("Enter the second number: ");
        int secondNumber = scanner.nextInt();

        int sum = firstNumber + secondNumber;
        System.out.println("The sum of " + firstNumber + " and " + secondNumber + " is: " + sum);
        scanner.close();
        return sum;
    }

    public static void main(String[] args){
        Sum sum = new Sum();
        sum.SumNumbers();
        System.out.println("Sum method executed successfully.");
    }
}